### PR TITLE
Add suspend option to in dummy plugin

### PR DESCRIPTION
**Pull request by cosmo0920**

### CODE REVIEW

**Comment by tagomoris** [comment]
Please leave `storage_local` as is, because these code will be fixed once more with plugin id auto-configuration (with root path by system config)

**Comment by tagomoris** [comment]
And, could you rebase changes of `in_dummy` on current master?

**Comment by cosmo0920** [comment]
I've rebased with current master.
But suspend option related tests don't pass yet.

**Comment by cosmo0920** [comment]
> Please leave storage_local as is, because these code will be fixed once more with plugin id auto-configuration (with root path by system config)

Oh.... I think almost storage_local code should be leave as-is. But this line https://github.com/fluent/fluentd/pull/900/files#diff-0380de12f73675c40904702aab31701cR61 should be fixed because without this fix I always got the following error:

```log
================================================================================
Error: test: value of auto increment key is suspended after stop-and-start(DummyTest::suspend internal counters if suspend is true): Fluent::ConfigError: Directory is not writable for plugin storage file 'test/tmp/in_dummy/store/json'
/home/hhatake/Github/fluentd/lib/fluent/plugin/storage_local.rb:61:in `configure'
/home/hhatake/Github/fluentd/lib/fluent/plugin_helper/storage.rb:92:in `block in configure'
/home/hhatake/Github/fluentd/lib/fluent/plugin_helper/storage.rb:87:in `each'
/home/hhatake/Github/fluentd/lib/fluent/plugin_helper/storage.rb:87:in `configure'
/home/hhatake/Github/fluentd/lib/fluent/plugin/in_dummy.rb:61:in `configure'
/home/hhatake/Github/fluentd/lib/fluent/test/driver/base.rb:78:in `configure'
test/plugin/test_in_dummy.rb:12:in `create_driver'
test/plugin/test_in_dummy.rb:163:in `block (2 levels) in <class:DummyTest>'
================================================================================
```

**Comment by tagomoris** [comment]
It's time to revoke this thread, and there is just a point to be fixed after many fixes during pending time.
@cosmo0920 Could you see my review comment (following this comment)?

File: lib/fluent/plugin/in_dummy.rb [hunk 1]
```diff
@@ -58,12 +60,16 @@ def initialize
     def configure(conf)
       super
       @dummy_index = 0
+      config = conf.elements.select{|e| e.name == 'storage' }.first
+      @storage = storage_create(usage: 'suspend', conf: config, type: :local)
```
Comment (tagomoris) [inline lib/fluent/plugin/in_dummy.rb]: Rebase this tree on master, and please use `default_type` instead of type keyword argument.

Comment (cosmo0920) [inline lib/fluent/plugin/in_dummy.rb]: OK. I'll try your suggested changes within a week.

**Comment by tagomoris** [comment]
LGTM!

**Comment by cosmo0920** [comment]
Thanks!!!

**Pull request merged**